bigquery_etl/shredder/delete.py

```diff
@@ -203,28 +203,59 @@ def delete_from_partition(
         job_config.write_disposition = bigquery.WriteDisposition.WRITE_TRUNCATE
 
     def create_job(client):
-        field_condition = " OR ".join(
-            f"""
-             {field} IN (
-               SELECT
-                 {source.field}
-               FROM
-                 `{sql_table_id(source)}`
-               WHERE
-            """
-            + " AND ".join((source_condition, *source.conditions))
-            + ")"
-            for field, source in zip(target.fields, sources)
-        )
-        query = reformat(
-            f"""
-            {"DELETE" if use_dml else "SELECT * FROM"}
-              `{sql_table_id(target)}`
-            WHERE
-              ({field_condition}){"" if use_dml else " IS NOT TRUE"}
-              AND {partition.condition}
-            """
-        )
+        if use_dml:
+            field_condition = " OR ".join(
+                f"""
+                 {field} IN (
+                   SELECT
+                     {source.field}
+                   FROM
+                     `{sql_table_id(source)}`
+                   WHERE
+                """
+                + " AND ".join((source_condition, *source.conditions))
+                + ")"
+                for field, source in zip(target.fields, sources)
+            )
+            query = reformat(
+                f"""
+                DELETE
+                  `{sql_table_id(target)}`
+                WHERE
+                  ({field_condition})
+                  AND {partition.condition}
+                """
+            )
+        else:
+            field_joins = "".join(
+                f"""
+                 LEFT JOIN (
+                   SELECT DISTINCT
```

Comment:
Since we're only keeping the rows where the join doesn't match this `DISTINCT` isn't technically necessary, and I'd be curious if removing the `DISTINCT` might make any difference in the query plan/performance.

Reply:
you're right. i was just being over-cautious about generating duplicate rows. I'll open an issue and investigate this next week.

Reply:
This SO discussion indicates that the DISTINCT is beneficial: https://stackoverflow.com/a/45471070 but it's 5 years old. Would be interesting to see results, but I don't think we end up with all that many dupes in the deletion request tables so might not make much of a difference.

Reply:
per https://github.com/mozilla/bigquery-etl/pull/2715#issuecomment-1034318779 removing the `DISTINCT` was marginally faster, either because of normal variability or because of the low number of duplicate IDs we see in the deletion request table makes `DISTINCT` inefficient.